legal-api/src/legal_api/services/permissions.py

```diff
@@ -0,0 +1,201 @@
+    DIRECTOR_CHANGE_FILING = 'DIRECTOR_CHANGE_FILING'
+    FIRM_CHANGE_FILING = 'FIRM_CHANGE_FILING'
+    FIRM_CONVERSION_FILING = 'FIRM_CONVERSION_FILING'
+    FIRM_DISSOLUTION_FILING = 'FIRM_DISSOLUTION_FILING'
+    INCORPORATION_APPLICATION_FILING = 'INCORPORATION_APPLICATION_FILING'
+    NOTICE_WITHDRAWAL_FILING = 'NOTICE_WITHDRAWAL_FILING'
+    RESTORATION_REINSTATEMENT_FILING = 'RESTORATION_REINSTATEMENT_FILING'
+    REGISTRATION_FILING = 'REGISTRATION_FILING'
+    SPECIAL_RESOLUTION_FILING = 'SPECIAL_RESOLUTION_FILING'
+    ADDRESS_CHANGE_FILING = 'ADDRESS_CHANGE_FILING'
+    
+class PermissionService:
+    """Service to manage permissions for user roles."""
+
+    @staticmethod
+    def get_authorized_permissions_for_user():
+        """
+        Returns a JSON response containing the authorized permissions for the current user.
+        """
+        authorized_role = PermissionService.get_authorized_user_role()
+        if not authorized_role:
+            return jsonify({
+                'message': 'No authorized role found.',
+                'authorizedPermissions': []
+            }), HTTPStatus.OK
+
+        cache_key = f'authorized_permissions_{authorized_role}'
+        cached_permissions = cache.get(cache_key)
+
+        if cached_permissions:
+            authorized_permissions = cached_permissions
+        else:
+            authorized_permissions = AuthorizedRolePermission.get_authorized_permissions_by_role_name(authorized_role)
+            cache.set(cache_key, authorized_permissions)
+
+        return authorized_permissions
+    
+    @staticmethod
+    def get_authorized_user_role() -> str:
+        """Return the first matching authorized role from the JWT, based on priority."""
+        from legal_api.services.authz import (
+    STAFF_ROLE, SBC_STAFF_ROLE, CONTACT_CENTRE_STAFF_ROLE, MAXIMUS_STAFF_ROLE, PUBLIC_USER
+    )
+        role_priority = [
+            STAFF_ROLE,
+            SBC_STAFF_ROLE,
+            CONTACT_CENTRE_STAFF_ROLE,
+            MAXIMUS_STAFF_ROLE,
+            PUBLIC_USER,
+        ]
+
+        token_info = getattr(g, 'jwt_oidc_token_info', {}) or {}
+
+        roles_in_token = token_info.get('realm_access', {}).get('roles', [])
+        for role in role_priority:
+            if role in roles_in_token:
+                return role
+        return None
+    
+    @staticmethod
+    def get_filing_permission_mapping():
+        """Return dictionary containing rules for filings are allowed for different roles."""
+        # pylint: disable=import-outside-toplevel
+        from legal_api.core.filing import Filing as CoreFiling
+
+        return {
+                CoreFiling.FilingTypes.TRANSITION.value:
+                ListFilingsPermissionsAllowed.TRANSITION_FILING.value
+                ,
+                CoreFiling.FilingTypesCompact.DISSOLUTION_VOLUNTARY.value:
+            ListFilingsPermissionsAllowed.VOLUNTARY_DISSOLUTION_FILING.value
```

Comment:
If this is a firm, check FIRM_DISSOLUTION_FILING permission.

Else, check VOLUNTARY_DISSOLUTION_FILING.

Reply:
Outdated.